packages/@aws-cdk/aws-iam/lib/private/precreated-role.ts

```diff
@@ -0,0 +1,305 @@
+  }
+
+  public grantPassRole(grantee: IPrincipal): Grant {
+    return this.role.grantPassRole(grantee);
+  }
+
+  public grantAssumeRole(identity: IPrincipal): Grant {
+    return this.role.grantAssumeRole(identity);
+  }
+
+  /**
+   * Resolve any references and replace with a more user friendly value. This is the value
+   * that will appear in the report, so instead of getting something like this (not very useful):
+   *
+   *     "Resource": {
+   *       "Fn::Join": [
+   *         "",
+   *         [
+   *           "arn:",
+   *           {
+   *             "Ref": "AWS::Partition"
+   *           },
+   *           ":iam::",
+   *           {
+   *             "Ref": "AWS::AccountId"
+   *           },
+   *           ":role/Role"
+   *         ]
+   *       ]
+   *     }
+   *
+   * We will instead get:
+   *
+   *     "Resource": "arn:(PARTITION):iam::(ACCOUNT):role/Role"
+   *
+   * Or if referencing a resource attribute
+   *
+   *     "Resource": {
+   *       "Fn::GetAtt": [
+   *         "SomeResource",
+   *         "Arn"
+   *       ]
+   *     }
+   *
+   * Becomes
+   *
+   *     "(Path/To/SomeResource.Arn)"
+   */
+  private resolveReferences(ref: any): any {
+    if (Array.isArray(ref)) {
+      return ref.map(r => this.resolveReferences(r));
+    } else if (typeof ref === 'object') {
+      return this.resolveJsonObject(ref);
+    }
+    const resolvable = Tokenization.reverseString(ref);
+    if (resolvable.length === 1 && Reference.isReference(resolvable.firstToken)) {
+      return `(${resolvable.firstToken.target.node.path}.${resolvable.firstToken.displayName})`;
+    } else {
+      const resolvedTokens = resolvable.mapTokens({
+        mapToken: (r: IResolvable) => {
+          if (Reference.isReference(r)) {
+            return `(${r.target.node.path}.${r.displayName})`;
+          }
+          const resolved = Stack.of(this).resolve(r);
+          if (typeof resolved === 'object' && resolved.hasOwnProperty('Ref')) {
+            switch (resolved.Ref) {
+              case 'AWS::AccountId':
+                return '(ACCOUNT)';
+              case 'AWS::Partition':
+                return '(PARTITION)';
+              case 'AWS::Region':
+                return '(REGION)';
+              default:
+                return r;
+            }
+          }
+          return r;
+        },
+      });
+      return resolvedTokens.join(new StringConcat());
+    }
+  }
+
+  private resolveJsonObject(statement: { [key: string]: any }): any {
+    const newStatement = statement;
+    for (const [key, value] of Object.entries(statement)) {
+      newStatement[key] = this.resolveReferences(value);
+    }
+    return newStatement;
+  }
+}
+
+/**
+ * Options for generating the role policy report
+ */
+interface RoleReportOptions {
+  /**
+   * The name of the IAM role.
+   *
+   * If this is not provided the role will be assumed
+   * to be missing.
+   *
+   * @default 'missing role'
```

Comment:
[nit] I think we could have more than one role with `missing role` name in the report.
Do you think the context offers enough information for the customer's report readability?